core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/XmlProtocolUnmarshaller.java

```diff
@@ -58,8 +61,16 @@ public static XmlProtocolUnmarshaller create() {
 
     public <TypeT extends SdkPojo> TypeT unmarshall(SdkPojo sdkPojo,
                                                     SdkHttpFullResponse response) {
-        XmlElement document = XmlResponseParserUtils.parse(sdkPojo, response);
-        return unmarshall(sdkPojo, document, response);
+
+        if (hasPayloadMembersOnUnmarshall(sdkPojo)
+            && !hasExplicitBlobPayloadMember(sdkPojo)
+            && !hasExplicitStringPayloadMember(sdkPojo)
+            && response.content().isPresent()) {
```

Comment:
This complex of a decision stuffed into an if statement on the hot code path has me questioning if there's a simpler way. I don't understand the situation well enough to answer that, but at minimum it might be worth giving a name to these 4 checks... `if (hasXmlPayload(sdkPojo)) {`?

If there isn't a better way, there may be a performance hit to the has* checks running on every response when the sdkPojo has a lot of fields. Traditional wisdom would be to benchmark before worrying about optimization, but it might be faster to just consider combining the 3 has* checks than taking the time to benchmark. That wouldn't be my suggestion if it made the code harder to understand, but I don't think that's the case here.

Reply:
Good point, I was going off what we did in [`JsonProtocolUnmarshaller`](https://github.com/aws/aws-sdk-java-v2/blob/master/core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/unmarshall/JsonProtocolUnmarshaller.java#L193)

Will refactor to combine the 3 has* checks, here and in `JsonProtocolUnmarshaller`